randobot/handler.py

```diff
@@ -161,13 +173,426 @@ async def race_data(self, data):
+                    'drafted_settings': {
+                        'bans': {},
+                        'picks': {},
+                        'data': {}
+                    },
+                })
+            elif args[0] == 'qualifier' and not draft.get('enabled'):
+                # Restrict qualifier argument to Race Moderators.
+                if not can_moderate(message):
+                    return
+                draft.update({
+                    'enabled': True,
+                    'race_type': args[0],
+                    'available_settings': self.zsr.load_available_settings(),
+                    'drafted_settings': {
+                        'picks': {},
+                        'data': {}
+                    }
+                })
+                await self.send_message(
+                    'Welcome to OoTR Draft Mode! '
+                    'You can disable Draft Mode with !s7 cancel.'
+                ),
+                await self.send_message(
+                    f"This is a {args[0]} race. Race monitors, use !seed 15 minutes prior to race start for a seed."
+                )
+            elif args[0] in ('tournament', 'practice', 'qualifier') and draft.get('enabled'):
+                await self.send_message(
+                    'Draft Mode is already enabled.'
+                )
+            elif args[0] == 'cancel':
+                if draft.get('enabled'):
+                    if draft.get('race_type') == 'tournament':
+                        # Only allow Race Moderators to cancel once drafting is complete.
+                        if draft.get('status') == 'complete' and not can_moderate(message):
+                            await self.send_message(
+                                'Drafting already complete. Contact a Race Moderator for assistance.'
+                            )
+                            return
+                        await self.ex_fpa(['off'], message),
+                    # Only allow Race Moderators to cancel qualifier races.
+                    elif draft.get('race_type') == 'qualifier':
+                        if not can_moderate(message):
+                            return
+                    await self.send_message(
+                        'Draft Mode has been disabled.'
+                    )
+                    draft.clear()
+                    return
+                await self.send_message(
+                    'Draft Mode is currently disabled.'
+                )
+
+    async def ex_first(self, args, message):
+        """
+        Handle !first commands.
+
+        Allow higher-seeded player to pick first.
+        """
+        draft = self.state.get('draft_data')
+        if self._race_in_progress() or not draft.get('status') == 'select_order':
+            return
+        
+        reply_to = message.get('user', {}).get('name')
+        racer = draft.get('racers')
+
+        # Compare sender to draft_data 
+        if not racer[0].get('name') == reply_to:
```

Comment:
I changed `determine_higher_seed()` to compare ids instead of names and if those match, grab the name and add it to the list `entrants`. That should hopefully mitigate this issue